core/server/router.go

```diff
@@ -70,6 +70,8 @@ func (rg *router) Handle(pattern string, handler http.Handler) {
 	// When this route is set we mark the rootSet as true
 	rg.rootSet = rg.rootSet || (pattern == "/")
 
+	handler = recoverer(handler)
```

Comment:
@tatang26 I know there is a technical explanation for this but this line seems to hide the recoverer as a middleware inside this method.